pkg/controller/replication/replication_controller.go

```diff
@@ -386,31 +386,32 @@ func (rm *ReplicationManager) enqueueControllerAfter(obj interface{}, after time
 // worker runs a worker thread that just dequeues items, processes them, and marks them done.
 // It enforces that the syncHandler is never invoked concurrently with the same key.
 func (rm *ReplicationManager) worker() {
-	workFunc := func() bool {
-		key, quit := rm.queue.Get()
-		if quit {
-			return true
-		}
-		defer rm.queue.Done(key)
-
-		err := rm.syncHandler(key.(string))
-		if err == nil {
-			rm.queue.Forget(key)
-			return false
-		}
-
-		rm.queue.AddRateLimited(key)
-		utilruntime.HandleError(err)
-		return false
-	}
 	for {
```

Comment:
Can you follow the pattern used in the replica set controller and split this out into a separate func? Also invert the returned booleans.